Tutorial translation

pwdtnx opened the pull request:
I added a tutorial section by translating [Deno で Vim/Neovim のプラグインを書く (denops.vim)](https://zenn.dev/lambdalisue/articles/b4a31fba0b1ce95104c9) to English.

pwdtnx committed:
add a tutorial in Japanese written by Alisue
https://zenn.dev/lambdalisue/articles/b4a31fba0b1ce95104c9

pwdtnx committed:
translate tutorial from Japanese to English

pwdtnx committed:
divide sections into indivisual pages

lambdalisue reviewed (changes requested):
It seems this PR is more complete than #3 so I would merge this one but please

1. Use `Neovim` instead of `NeoVim`
2. Please re-arrange commits. `update` or such kind of commit message does not explain the commit itself

Additionally, I haven't reviewed it all yet. I'll re-review it once you've fixed the above issues.

pwdtnx commented:
@lambdalisue 
Thank you for looking out for this PR.
I have corrected misspellings and squashed commits that had been pointed out:
> 1. Use `Neovim` instead of `NeoVim`
> 2. Please re-arrange commits. `update` or such kind of commit message does not explain the commit itself

lambdalisue commented on src/tutorial.md:
```diff
@@ -1,122 +1,106 @@
-| Denops プラグイン | Denops を用いて書かれた Vim/Neovim 双方で動作するプラグインを表します   |
+| Term                  | Meaning                                                                     |
+| --------------------- | --------------------------------------------------------------------------- |
+| vim                   | Vim or NeoVim.                                                          |
```
**must**

Not NeoVim but Neovim. Please replace all NeoVim to Neovim.

lambdalisue replied:
Fixed

lambdalisue commented on src/tutorial.md:
````diff
@@ -1,122 +1,106 @@
 ```
 
-なお、既にインストール済みであれば、以下のコマンドで最新版にアップデートしてください。
+If you have already installed Deno, update it to the latest version.
````
may

Let use "upgrade" rather than "update" while the command below says "upgrade" instead.

pwdtnx replied:
Corrected at af3096c.

lambdalisue commented on src/tutorial/adding-a-skelton-of-denops-plugin.md:
```diff
@@ -0,0 +1,23 @@
+
+Then you restart vim, and you can see a message `[denops] Hello Denops!` on the vim window.
+
+![](https://storage.googleapis.com/zenn-user-upload/y8fyu6tap3tapjbxtltni6jynq8y)
```
**must**

Please do NOT refer to images on Zenn directly. Instead, save images under `assets/img/...` or whatever and refer to that.

lambdalisue replied:
Please use https://github.com/vim-denops/denops-documentation/issues/6 issue to paste screenshots.

pwdtnx replied:
Saved images under `src/img/...` at 2138405.

lambdalisue commented on src/tutorial/adding-an-api.md:
````diff
@@ -0,0 +1,40 @@
+```
+
+Note that you can register a function that satisfies the following as an API:
+- All of its arguments must be `unknown`.
````
may

```suggestion
- All of its argument types must be `unknown`.
```

pwdtnx replied:
I think it would be better to use "of type" for type restrictions, so revised in this way at af3096c.

lambdalisue commented on src/tutorial/developing-more-applicative-plugin.md:
```diff
@@ -0,0 +1,105 @@
+So let me ask you, out of the blue, have you ever itched to solve mazes while programming?
+I never have.
+In any case, there may be people who love solving mazes and can't get enough of it.
+So let's try to develop a Denops plugin that can generate and display a maze in vim at any time.
```
Cool 🤣

lambdalisue commented:
https://github.com/vim-denops/denops-documentation/commit/7ea495ad87833bd6bb63bdf00c5ad413b7454170

I've added a commit to ignore changes under `./docs` so please rebase from the current main 'nd force push.

lambdalisue commented:
Hum... I thought 7ea495a works... Ok, never mind. Please fix the issues I've pointed.

pwdtnx committed:
save images on articles in the repository

pwdtnx committed:
revised expressions pointed out in the review

lambdalisue reviewed (approved):
lgtm

pwdtnx committed:
fixed misspellings and broken link

lambdalisue reviewed (approved):
LGTM

lambdalisue commented:
I'll merge it tonight or wait for @monaqa reaction while he made similar PR on #3

monaqa commented on src/tutorial/developing-more-applicative-plugin.md:
````diff
@@ -0,0 +1,105 @@
+};
+```
+
+In this code, `denops.cmd()` calls the `enew` command of vim to open a new buffer in the current window and then `denops.call()` calls `setline()` to write the maze to the buffer.
````
Suggestion

How about this translation?

> ... and then `denops.call()` calls the `setline()` function to write the maze ...

This contrast makes it clearer that `denops.cmd()` executes Vim *commands*, while `denops.call()` calls Vim *functions*.

pwdtnx replied:
Thank you for your review and I agree with your intension to clearify difference of the meanings between Vim _commands_ and Vim _functions_.
So I've revised as in 882ddc9.

monaqa commented:
I think it's perfectly fine, except for one small suggestion.

Thank you for the excellent translation :)

pwdtnx committed:
revised some expressions and styles

pwdtnx commented:
@lambdalisue 
I'm sorry I've made some additional corrections for misspells and styles in spite of the completion of your review.

monaqa reviewed (approved):
LGTM!

lambdalisue reviewed (approved):
Cool thanks

lambdalisue commented:
@pwdtnx @monaqa Merged. Thanks for your contributions 🎉
